src/System.Private.CoreLib/src/System/ByReference.cs

```diff
@@ -23,7 +23,7 @@ public ByReference(ref T value)
             // Implemented as a JIT intrinsic - This default implementation is for 
             // completeness and to provide a concrete error if called via reflection
             // or if intrinsic is missed.
-            throw new NotSupportedException();
```

Comment:
Is it OK to change exception type for corert here and below?

Reply:
Yes, it should be fine. I would add `using System` at the top of the file so that this can be without `System.`